src/eval-frame/components/reps/rep-utils/tiny-rep-serializer.js

```diff
@@ -0,0 +1,80 @@
+export function getClass(obj) {
+  if (obj === null) return "Null";
+  if (obj === undefined) return "Undefined";
+  return obj.constructor.name;
+}
+
+export const getType = obj =>
+  Object()
+    .toString.call(obj)
+    .split(" ")[1]
+    .slice(0, -1);
+
+const typesWithLength = [
+  "Array",
+  "String",
+  "Int8Array",
+  "Uint8Array",
+  "Uint8ClampedArray",
+  "Int16Array",
+  "Uint16Array",
+  "Int32Array",
+  "Uint32Array",
+  "Float32Array",
+  "Float64Array",
+  "Function",
+  "GeneratorFunction"
+];
+
+const typesWithByteLength = ["ArrayBuffer", "DataView"];
+
+const typesWithSize = ["Map", "Set"];
+
+export function objSize(obj) {
+  if (obj === null) return undefined;
+  if (obj === undefined) return undefined;
+  const type = getType(obj);
+  if (type === "RegExp") return obj.source.length;
+  if (typesWithLength.includes(type)) return obj.length;
```

Comment:
Could we just check whether `obj.length !== undefined` (i.e. duck type it)?  Then this would capture as of yet unknown classes.

Reply:
the problem with duck typing in this case is that a user i might very plausibly define e.g.:
```
couch = {height: 3, length:5, depth:2.5, color: "puce", material: "cotton"}
```
in fact, i think i may even have a notebook with objects that have `length` property used this way. i think we're stuck with this non-duck approach, since js `obj.length` has less convenient semantics than python `len(obj)` :-/

(my general sense is that a lot of the stuff around reps will require this kind of careful and fiddly hand-curation and enumeration)